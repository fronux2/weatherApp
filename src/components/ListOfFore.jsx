import { View, Text, StyleSheet, Image} from 'react-native'

export default function ListOfFore ({foreCast}) {
    return foreCast && (
        foreCast.map(data => {
          return (
          <View key={data.date_epoch} style={styles.container2}>
            <Text style={styles.text}>{data.date}          </Text>
            <Image style={styles.image} source={{ uri: `https:${data.day.condition.icon}` }}></Image>
            <Text style={styles.text}>{data.day.maxtemp_c}/</Text>
            <Text style={styles.text}>{data.day.mintemp_c}</Text>
          </View>
          )
        })
      )     
        
      
}
const styles = StyleSheet.create({
    image: {
        width: 64, // Ancho deseado de la imagen
        height: 64 // Altura deseada de la imagen
        },
    text: {
        color: 'white'
        },
    container2: {
    flex: 0,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
        }
})