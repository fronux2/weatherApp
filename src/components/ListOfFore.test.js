import ListOfFore from './ListOfFore.jsx'
/* import { render, screen } from '@testing-library/react-native' */
import renderer from 'react-test-renderer'
test('should render one element', () => {
  const forecast = [
    {
      date_apoch: 12345,
      date: '2012-02-02T12:00:00',
      day: {
        condition: {
          icon: 'sunny.png'
        },
        maxtemp_c: '77',
        mintemp_c: '33'
      }
    },
    {
      date_apoch: 54321,
      date: '2023-09-20T15:30:00',
      day: {
        condition: {
          icon: 'sunny.png'
        },
        maxtemp_c: '82',
        mintemp_c: '25'
      }
    }
  ]
  const component = renderer.create(<ListOfFore forecast={forecast} />).toJSON()
  expect(component).toMatchSnapshot()
})
