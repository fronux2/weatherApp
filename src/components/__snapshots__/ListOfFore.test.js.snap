// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one element 1`] = `null`;

exports[`should render two list items 1`] = `null`;
